Python/Tests/DebuggerTests/IpcJsonTests.cs

```diff
@@ -0,0 +1,312 @@
+// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PythonTools.Infrastructure;
+using Microsoft.PythonTools.Ipc.Json;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using TestUtilities;
+using TestUtilities.Python;
+
+namespace DebuggerTests {
+    [TestClass]
+    public class IpcJsonTests {
```

Comment:
AnalysisTests doesn't depend on Ipc.Json project (Analyzer projects depends on Ipc.Json, but not Analysis).  So I don't think that's the right place either.  It would probably be best to be in its own test project.

---
In reply to: [108000250](https://github.com/Microsoft/PTVS/pull/2323#discussion_r108000250) [](ancestors = 108000250,107991525)